dev-ostelco-ios-clientTests/CoordinatorTests/SingaporeUserHappyFlowWithSingPassStageDeciderTests.swift

```diff
@@ -66,128 +66,85 @@ class SingaporeUserHappyFlowWithSingPassStageDeciderTests: XCTestCase {
+        let region = RegionResponse(
+            region: Region(id: "sg", name: "Singapore"),
+            status: .PENDING,
+            simProfiles: nil,
+            kycStatusMap: KYCStatusMap(jumio: nil, myInfo: .PENDING, nricFin: .APPROVED, addressPhone: nil)
+        )
         
-        XCTAssertEqual(decider.compute(context: context, localContext: localContext), .regionOnboarding)
+        XCTAssertEqual(decider.stageForRegion(region: region, localContext: localContext), .verifyMyInfo(code: "xxx"))
     }
     
     func testUserHasCompletedSingpassAndVerifiedTheirAddress() {
         let decider = StageDecider()
         let localContext = LocalContext(hasSeenNotificationPermissions: true, hasSeenLocationPermissions: true)
-        
-        let context = Context(
-            customer: CustomerModel(id: "xxx", name: "xxx", email: "xxxx@gmail.com", analyticsId: "xxxx", referralId: "xxxx"),
-            regions: [
-                RegionResponse(
-                    region: Region(id: "sg", name: "Singapore"),
-                    status: .APPROVED,
-                    simProfiles: nil,
-                    kycStatusMap: KYCStatusMap(jumio: .PENDING, myInfo: .APPROVED, nricFin: .PENDING, addressPhone: .PENDING)
-                )
-            ]
+        let region = RegionResponse(
+            region: Region(id: "sg", name: "Singapore"),
+            status: .APPROVED,
+            simProfiles: nil,
+            kycStatusMap: KYCStatusMap(jumio: .PENDING, myInfo: .APPROVED, nricFin: .PENDING, addressPhone: .PENDING)
         )
         
-        XCTAssertEqual(decider.compute(context: context, localContext: localContext), .eSimOnboarding)
+        XCTAssertEqual(decider.stageForRegion(region: region, localContext: localContext), .eSimOnboarding)
     }
     
     func testUserHasSeenTheESimOnboarding() {
         let decider = StageDecider()
         let localContext = LocalContext(hasSeenNotificationPermissions: true, hasSeenESimOnboarding: true)
-        
-        let context = Context(
-            customer: CustomerModel(id: "xxx", name: "xxx", email: "xxxx@gmail.com", analyticsId: "xxxx", referralId: "xxxx"),
-            regions: [
-                RegionResponse(
-                    region: Region(id: "sg", name: "Singapore"),
-                    status: .APPROVED,
-                    simProfiles: nil,
-                    kycStatusMap: KYCStatusMap(jumio: .PENDING, myInfo: .APPROVED, nricFin: .PENDING, addressPhone: .PENDING)
-                )
-            ]
+        let region = RegionResponse(
+            region: Region(id: "sg", name: "Singapore"),
+            status: .APPROVED,
+            simProfiles: nil,
+            kycStatusMap: KYCStatusMap(jumio: .PENDING, myInfo: .APPROVED, nricFin: .PENDING, addressPhone: .PENDING)
```

Comment:
This snippet of code (from liner 127 to 131) is repeated multiple times.  Is there a reason why it is not delegated to a method?

Reply:
It's not always the same value for each, and embedding it into some shared value impedes the speed of working with tests. This way you can clearly see what values the test is considering and change them without chasing changes in other tests.